apps/client/src/logging/cloud-watch-logger.ts

```diff
@@ -0,0 +1,164 @@
+import { type Log, type Logger } from '@iot-app-kit/core';
+import {
+  CloudWatchLogsClient,
+  CreateLogStreamCommand,
+  PutLogEventsCommand,
+} from '@aws-sdk/client-cloudwatch-logs';
+import throttle from 'lodash/throttle';
+import { type AuthService } from '~/auth/auth-service.interface';
+import { authService } from '~/auth/auth-service';
+
+// at most one putLogEvents() call every X ms
+export const PUT_LOG_EVENTS_THROTTLE = 3000;
+
+export enum LOG_LEVEL {
+  log = 30,
+  warn = 40,
+  error = 50,
+}
+
+interface LogEvent {
+  message: string;
+  timestamp: number;
+}
+
+class CloudWatchLogsQueue {
+  private readonly queue = new Set<LogEvent>();
```

Comment:
Curious: how do you make sure that their is only one queue ? i.e singleton

Reply:
`CloudWatchLogsQueue` is an internal class that only `CloudWatchLogger` has access to; designed to have 1 queue per CloudWatchLogger instance